### PR TITLE
chore(textproto): minor cleanups

### PR DESCRIPTION
**Pull request by iuioiua**
1. Removes seemingly obsolete `[textproto] Reader : DotBytes` test, closing #2551.
2. Re-enables `[textproto] #4521 issue` test
3. Deletes the deprecated `readMIMEHeaders()`

### CODE REVIEW

File: collections/zip.ts [hunk 1]
```diff
@@ -27,8 +27,7 @@ import { minOf } from "./min_of.ts";
 export function zip<T extends unknown[]>(
   ...arrays: { [K in keyof T]: T[K][] }
 ): T[] {
-  let minLength = minOf(arrays, (it) => it.length);
-  if (minLength === undefined) minLength = 0;
+  const minLength = minOf(arrays, (it) => it.length) ?? 0;
```
Comment (kt3k) [inline collections/zip.ts:30]: 👍

**Comment by kt3k** [comment]
> Removes seemingly obsolete [textproto] Reader : DotBytes test, closing https://github.com/denoland/deno_std/issues/2551.
Re-enables [textproto] #4521 issue test

These changes look good to me.

> Deletes the deprecated readMIMEHeaders()

I think we should keep this a little bit longer to better communicate with users about deprecation. We just deprecated it in 0.154.0. I think it's a little too quick to delete them in 0.155.0. ([vercel-deno](https://github.com/vercel-community/deno), [servest](https://github.com/keroxp/servest), etc seem using this)

**Comment by iuioiua** [comment]
> > Removes seemingly obsolete [textproto] Reader : DotBytes test, closing #2551.
> > Re-enables [textproto] #4521 issue test
> 
> These changes look good to me.
> 
> > Deletes the deprecated readMIMEHeaders()
> 
> I think we should keep this a little bit longer to better communicate with users about deprecation. We just deprecated it in 0.154.0. I think it's a little too quick to delete them in 0.155.0. ([vercel-deno](https://github.com/vercel-community/deno), [servest](https://github.com/keroxp/servest), etc seem using this)

Fair enough. Is there a version when this is able to be removed? I'd like to add the comment for future reference. If you're not sure, we can just skip over it.

**Comment by kt3k** [comment]
> Is there a version when this is able to be removed? I'd like to add the comment for future reference. If you're not sure, we can just skip over it.

We don't have a rule for this yet, but I feel 4~5 minor versions work fine. (Note: We usually release a new std minor version every week, and new Deno minor version every month.)

**Review by kt3k — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**